src/kOS/Suffixed/AggregateResourceValue.cs

```diff
@@ -1,44 +1,41 @@
 ﻿using System.Collections.Generic;
 using kOS.Safe.Encapsulation;
+using kOS.Safe.Encapsulation.Suffixes;
+using kOS.Suffixed.Part;
 
 namespace kOS.Suffixed
 {
     public class AggregateResourceValue : Structure
     {
         private readonly string name;
+        private readonly SharedObjects shared;
         private double amount;
         private double capacity;
-        private readonly ListValue parts;
+        private readonly SuffixedList<PartValue> parts;
```

Comment:
Fair enough.  And like I said I'm in favor of the idea as a whole, but it requires a bit more major a refactor than this to really work right for its intended goal.

I do wonder if there may be some other way to design it that would make it possible to actually implement the idea in a "slow" refactor that can be done piecemeal but without needing to remove work.

I am in favor of compile-time enforcement of making it impossible to give a kerbalscript program access to a non-kOS underlying object.  That's why I was thinking of places like the variable dictionary and the stack as being the natural places to start.